code/modules/mob/living/basic/space_fauna/giant_spider/spider_abilities/lay_eggs.dm

```diff
@@ -0,0 +1,84 @@
+/datum/action/lay_eggs
+	name = "Lay Eggs"
+	desc = "Lay a cluster of eggs, which will soon grow into a normal spider."
+	button_icon = 'icons/mob/actions/actions_animal.dmi'
+	button_icon_state = "lay_eggs"
+	background_icon_state = "bg_alien"
+	overlay_icon_state = "bg_alien_border"
+	check_flags = AB_CHECK_CONSCIOUS
+	///How long it takes for a broodmother to lay eggs.
+	var/egg_lay_time = 12 SECONDS
+	///The type of egg we create
+	var/egg_type = /obj/effect/mob_spawn/ghost_role/spider
+
+/datum/action/lay_eggs/Grant(mob/grant_to)
+	. = ..()
+	if (!owner)
+		return
+	RegisterSignals(owner, list(COMSIG_MOVABLE_MOVED, COMSIG_DO_AFTER_BEGAN, COMSIG_DO_AFTER_ENDED), PROC_REF(update_status_on_signal))
+
+/datum/action/lay_eggs/Remove(mob/removed_from)
+	. = ..()
+	UnregisterSignal(removed_from, list(COMSIG_MOVABLE_MOVED, COMSIG_DO_AFTER_BEGAN, COMSIG_DO_AFTER_ENDED))
+
+/datum/action/lay_eggs/IsAvailable(feedback = FALSE)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(!isspider(owner))
+		return FALSE
```

Comment:
```suggestion
	if(!. || !isspider(owner))
		return FALSE
```
why not do something like this?

Reply:
I actually meant to remove that check and missed this one, it doesn't have a useful _functional_ reason to exist at this point and while it's only used by spiders right now there doesn't seem to be a reason to enforce it.